### PR TITLE
Update extension package to 3.1.0

### PR DESCRIPTION
**Pull request by timotheeguerin**
Update package to contain yarn -> npm migration to fix core package install

### CODE REVIEW

**Review by daviwil — APPROVED** [review]
Thanks Tim!

**Comment by daviwil** [comment]
Looks like the lockfile needs to be refreshed, that'll help get the CI bumped to start again.

File: packages/apps/autorest/package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "static-link": "^0.3.0",
     "@autorest/core": "~3.0.0",
     "@azure-tools/async-io": "~3.0.0",
-    "@azure-tools/extension": "~3.0.0",
+    "@azure-tools/extension": "~3.1.0",
```
Comment (daviwil) [inline packages/apps/autorest/package.json:58]: I believe this should be a real dependnecy of AutoRest since it gets used as part of the runtime code.

Comment (timotheeguerin) [reply packages/apps/autorest/package.json:58]: yeah, i am still having issues, I think bundledDependencies is acting weird with pnpm

Comment (daviwil) [reply packages/apps/autorest/package.json:58]: Oh great...  I hadn't considered that possibility

**Pull request closed without merging**